src/derivative_wrappers.jl

```diff
@@ -20,45 +20,58 @@ function derivative!(df::AbstractArray{<:Number}, f, x::Union{Number,AbstractArr
     nothing
 end
 
+jacobian_autodiff(f,x,_,_)=ForwardDiff.derivative(f,x)
+jacobian_autodiff(f,x::AbstractArray,_,hascolor::Type{Val{false}})=ForwardDiff.jacobian(f,x)
+function jacobian_autodiff(f,x::AbstractArray,integrator,hascolor::Type{Val{true}})
+  colorvec=integrator.f.colorvec
+  sparsity=integrator.f.jac_prototype
+  jac=integrator.f.jac
+  J=jac isa SparseMatrixCSC ? similar(jac) : zeros(size(jac)...)
```

Comment:
No need for splatting here, it seems.

Reply:
Sorry, what's splatting?

Reply:
Splatting is the `...`, and it leads to bad interactions with the compiler. It should be avoided whenever possible. Here it's not necessary.

Reply:
Oh, find an error. `jac` should be `f.jac_prototype` instead of `f.jac` which is a function. I was thinking about `cache.J` since `cache.J` sometimes is different from `jac_prototype` , but looks like it is not available here. I need to fix OrdDiffEq as well.

Reply:
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/836